libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1835,16 +1829,17 @@ bool PbftManager::pushPbftBlock_(PeriodData &&period_data, std::vector<std::shar
     pbft_chain_->updatePbftChain(pbft_block_hash, null_anchor);
   }
 
-  // Advance pbft consensus period
-  advancePeriod();
-
   LOG(log_nf_) << "Pushed new PBFT block " << pbft_block_hash << " into chain. Period: " << pbft_period
                << ", round: " << getPbftRound();
 
-  finalize_(std::move(period_data), std::move(dag_blocks_order));
+  finalize_(std::move(period_data), std::move(dag_blocks_order), synchronous_processing);
 
   db_->savePbftMgrStatus(PbftMgrStatus::ExecutedBlock, true);
   executed_pbft_block_ = true;
+
+  // Advance pbft consensus period
+  advancePeriod();
```

Comment:
maybe this change should be enough? 
If I understand correctly what is happening now, we won't be waiting indefinitely in `updateDposState_` because finalization is occuring in other thread. So when block would be finalized, we will get new values inside of `updateDposState_` and move on